workflow-manager/main.go

```diff
@@ -111,16 +110,78 @@ func work(inputBucket, k8sNS string) error {
 		names = append(names, attrs.Name)
 	}
 
-	var readybatches []*batch
-	var unreadybatches []*batch
+	var readyBatches []string
 	for k, v := range batches {
 		if v.metadata && v.avro && v.sig {
 			log.Printf("ready: %s", k)
-			readybatches = append(readybatches, v)
+			readyBatches = append(readyBatches, k)
 		} else {
 			log.Printf("unready: %s", k)
-			unreadybatches = append(unreadybatches, v)
 		}
 	}
+
+	log.Printf("starting %d jobs", len(readyBatches))
+	for _, batchName := range readyBatches {
+		err = startJob(ctx, clientset, batchName)
+		if err != nil {
+			return fmt.Errorf("starting job for batch %q: %w", batchName, err)
+		}
+	}
+	return nil
+}
+
+func startJob(ctx context.Context, clientset *kubernetes.Clientset, batchName string) error {
+	log.Printf("starting job for batch %q...", batchName)
+	jobName := fmt.Sprintf("i-batch-%s", strings.ReplaceAll(batchName, "/", "-"))
+
+	pathComponents := strings.Split(batchName, "/")
+	batchID := pathComponents[len(pathComponents)-1]
+	batchDate := strings.Join(pathComponents[0:len(pathComponents)-1], "/")
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobName,
+			Namespace: *k8sNS,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: "Never",
+					Containers: []corev1.Container{
+						{
+							Args: []string{
+								"batch-intake",
+								"--s3-use-credentials-from-gke-metadata",
+								"--aggregation-id", "fake-1",
+								"--batch-id", batchID,
+								"--date", batchDate,
+								"--ingestion-bucket", "gs://tbd",
+								"--validation-bucket", "gs://tbd",
+								"--ingestor-public-key", "what_is_my_pubkey",
+								"--ecies-private-key", "what_is_my_privkey",
+								"--share-processor-private-key", "what_is_my_other_privkey",
```

Comment:
These parameter names are out of date as of #89. In any case, since facilitator can accept these keys as arguments or environment variables, I see two ways to provide batch signing keys to the facilitator:
1. Plumb them into the workflow-manager process via environment variables or args, and then have workflow-manager pass them into the facilitator
2. Have the workflow-manager construct the Go API equivalent of this Terraform block:
```
env {
    name = "BATCH_SIGNING_PRIVATE_KEY"
    value_from {
        secret_key_ref {
            name = $(batch_signing_key_secret_name)
            key  = "signing_key"
        }
    }
}
```
(2) is a little nicer than (1) because it avoids ever exposing the secret to the workflow-manager's address space.

Providing the packet decryption key(s) is trickier, because facilitator needs all available private keys provided as a comma separated list (it needs all available keys provided because we can't tell up front what keys individual packets in a batch will have been encrypted to). We could rig up the Terraform side to provide all the decryption keys to workflow-manager, then have workflow-manager construct the comma-separated list. Also, for Narnia, we can take a shortcut and assume there will only be a single available packet decryption key and thus plug that secret in as an environment variable.